control-plane-operator/controllers/hostedcontrolplane/kas_pki_setup.go

```diff
@@ -0,0 +1,194 @@
+package hostedcontrolplane
+
+import (
+	"context"
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/upsert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type signerReconciler func(*corev1.Secret, config.OwnerRef) error
+type subReconciler func(target, ca *corev1.Secret, ownerRef config.OwnerRef) error
+
+func (r *HostedControlPlaneReconciler) setupKASClientSigners(
+	ctx context.Context,
+	hcp *hyperv1.HostedControlPlane,
+	p *pki.PKIParams,
+	createOrUpdate upsert.CreateOrUpdateFN,
+) error {
+	reconcileSigner := func(s *corev1.Secret, reconciler signerReconciler) (*corev1.Secret, error) {
+		applyFunc := func() error {
+			return reconciler(s, p.OwnerRef)
+		}
+
+		if _, err := createOrUpdate(ctx, r, s, applyFunc); err != nil {
+			return nil, fmt.Errorf("failed to reconcile secret '%s/%s': %v", s.Namespace, s.Name, err)
+		}
+		return s, nil
+	}
+
+	reconcileSub := func(target, ca *corev1.Secret, reconciler subReconciler) (*corev1.Secret, error) {
+		applyFunc := func() error {
+			return reconciler(target, ca, p.OwnerRef)
+		}
+
+		if _, err := createOrUpdate(ctx, r, target, applyFunc); err != nil {
+			return nil, fmt.Errorf("failed to reconcile secret '%s/%s': %v", target.Namespace, target.Name, err)
+		}
+		return target, nil
```

Comment:
Note: Using this pattern might be generally useful for the whole 3k lines `hostedcontrolplane_controller.go`. I think we could create
```go
constructSignerReconciler(
    ctx context.Context,
    r *HostedControlPlaneReconciler,
    p *pki.Params,
) func(s *corev1.Secret, reconcileFN signerReconciler) (*corev1.Secret, error)
```

(and similar to leaf certs) to avoid much of the duplicated code.

That should probably be done outside this PR, though.